### PR TITLE
fix: handle HTTPS request throw proxy

### PR DESCRIPTION
**Pull request by ludosch**
Work with and without proxy compagny

### CODE REVIEW

**Comment by kellyselden** [comment]
Can you provide a little more detail as to the problem please?

**Comment by ludosch** [comment]
Oh yes, HTTPS request throw proxy compagny can't reach target (native behaviour of nodejs), so, I can't use the ember-cli-update app, because it's fail when it try to access to codemods list (in json).

URL: https://github.com/ember-cli/ember-cli-update/blob/006239a5658eb7ac45d2978dc71bdc4c773f543b/src/index.js#L152

File: src/get-codemods.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 const https = require('https');
+const HttpProxyAgent = require('https-proxy-agent');
 
 module.exports = async function getCodemods(url) {
   let manifest = '';
 
   await new Promise((resolve, reject) => {
-    https.get(url, res => {
+
+    let proxy = process.env.https_proxy
```
Comment (kellyselden) [inline src/get-codemods.js]: Is this a convention of some sort?

Comment (ludosch) [reply src/get-codemods.js]: Yes, https_proxy et http_proxy are common use. You can see examples in tests of "node-https-agent" here : 
https://github.com/TooTallNate/node-https-proxy-agent/blob/master/test/test.js

Comment (ludosch) [reply src/get-codemods.js]: Does something missing ?

Comment (kellyselden) [reply src/get-codemods.js]: nope, looks good

**Comment by kellyselden** [comment]
Thanks for doing this!

**Pull request merged**